src/connector/postgres.rs

```diff
@@ -0,0 +1,402 @@
+                            .map(Value::from)
+                            .collect();
+
+                        Value::Array(val)
+                    }
+                    None => Value::Null,
+                },
+                PostgresType::NUMERIC_ARRAY => match row.try_get(i)? {
+                    Some(val) => {
+                        let val: Vec<Decimal> = val;
+
+                        let val: Vec<Value> = val
+                            .into_iter()
+                            .map(|d| d.to_string())
+                            .map(|s| s.parse::<f64>().unwrap())
+                            .map(|f| Number::from_f64(f).unwrap())
+                            .map(Value::Number)
+                            .collect();
+
+                        Value::Array(val)
+                    }
+                    None => Value::Null,
+                },
+                PostgresType::TEXT_ARRAY | PostgresType::NAME_ARRAY | PostgresType::VARCHAR_ARRAY => {
+                    match row.try_get(i)? {
+                        Some(val) => {
+                            let val: Vec<&str> = val;
+                            Value::Array(val.into_iter().map(Value::from).collect())
+                        }
+                        None => Value::Null,
+                    }
+                }
+                */
+                _ => match row.try_get(i)? {
+                    Some(val) => ParameterizedValue::Text(val),
+                    None => ParameterizedValue::Null,
+                },
+            };
+
+            Ok(result)
+        }
+
+        let mut row = ResultRow::default();
+
+        for i in 0..self.columns().len() {
+            row.values.push(convert(self, i)?);
+        }
+
+        Ok(row)
+    }
+}
+
+impl ToColumnNames for PostgresStatement {
+    fn to_column_names<'b>(&'b self) -> ColumnNames {
+        let mut names = ColumnNames::default();
+
+        for column in self.columns() {
+            names.names.push(String::from(column.name()));
+        }
+
+        names
+    }
+}
+
+impl PostgreSql {
+    pub fn new(config: Config, connections: u32) -> QueryResult<PostgreSql> {
+        let mut tls_builder = TlsConnector::builder();
+        tls_builder.danger_accept_invalid_certs(true); // For Heroku
+        let tls = MakeTlsConnector::new(tls_builder.build()?);
+
+        let manager = PostgresConnectionManager::new(config, tls);
+        let pool = r2d2::Pool::builder().max_size(connections).build(manager)?;
+
+        Ok(PostgreSql { pool })
+    }
+
+    fn with_connection_internal<F, T>(&self, f: F) -> QueryResult<T>
+    where
+        F: FnOnce(&mut PostgresConnection) -> QueryResult<T>,
+    {
+        let mut client = self.pool.get()?;
+        let result = f(&mut client);
+        result
+    }
+}
+
+impl From<postgres::error::Error> for Error {
+    fn from(e: postgres::error::Error) -> Error {
+        dbg!(&e);
+        // TODO: Ask J how to map to Failure::Error
```

Comment:
Error Handling: postgres::error::Error does not use the failure crate. What would be a good way to convert here?

Reply:
https://github.com/prisma/prisma/blob/alpha/server/prisma-rs/query-engine/connectors/sql-connector/src/error.rs#L242

The Failure crate provides the needed `From`.

Reply:
Done.